datumaro/components/hl_ops.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from typing import Dict, Optional, Union
+import os
+import os.path as osp
+import shutil
+
+from datumaro.components.converter import Converter
+from datumaro.components.dataset import (
+    Dataset, DatasetItemStorageDatasetView, IDataset,
+)
+from datumaro.components.dataset_filter import (
+    XPathAnnotationsFilter, XPathDatasetFilter,
+)
+from datumaro.components.environment import Environment
+from datumaro.components.extractor import Transform
+from datumaro.components.launcher import Launcher, ModelTransform
+from datumaro.components.operations import ExactMerge
+from datumaro.components.validator import TaskType, Validator
+from datumaro.util import parse_str_enum_value
+from datumaro.util.scope import on_error_do, scoped
+
+
+def transform(dataset: IDataset, method: Union[str, Transform], *,
+        env: Optional[Environment] = None, **kwargs) -> IDataset:
+    """
+    Applies some function to dataset items.
+
+    Results are computed lazily, if the transform supports this.
+
+    Args:
+        dataset (IDataset) - The dataset to be transformed
+        method (str or Transform) - The transformation function to be
+            applied to the dataset. If a string is passed, it is treated
+            as a plugin name, which is searched for in the environment
+            set by the 'env' argument
+        env (Environment) - A plugin collection. If not set, the default one
+            is used
+        **kwargs - Optional parameters for the transformation
+
+    Returns: a lazy-computible wrapper around the input dataset
+    """
+
+    if isinstance(method, str):
+        if env is None:
+            env = Environment()
+        method = env.transforms[method]
+
+    produced = method(dataset, **kwargs)
+
+    return Dataset(source=produced, env=env)
+
+def filter(dataset: IDataset, expr: str, *, #pylint: disable=redefined-builtin
+        filter_annotations: bool = False,
```

Comment:
IMO, the fact that different values of this function select completely different implementations suggests that this function should be split into two, `filter_items` and `filter_annotations`.

Reply:
It was done so some time ago, but it wasn't very comfortable to use. I'll reconsider such option.